### PR TITLE
[ADVAPP-697]: Improve processing of long AI responses by creating a "continue generating" button and process

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-697

### Technical Description

Moves to an event based streaming system to handle more complex streaming events, such as incomplete responses and mid-stream errors.

Encodes response content in base64 in order to avoid issues parsing events in JSON.

### Any deployment steps required?

No

### Are any Feature Flags Added?

No

_______________________________________________

### CODE REVIEW

File: app-modules/ai/src/Actions/CompleteResponse.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        $aiService->ensureAssistantAndThreadExists($thread);
+
+        return $aiService
+            ->completeResponse(
+                response: $response,
+                files: $thread->messages()
+                    ->whereNotNull('user_id')
+                    ->latest()
+                    ->first()
```
Comment (Orrison) [inline app-modules/ai/src/Actions/CompleteResponse.php:83]: Don't you already have this as `$response` above?

File: app-modules/ai/tests/Feature/Http/Controllers/CompleteResponseControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+use AdvisingApp\Ai\Exceptions\AiThreadLockedException;
+use AdvisingApp\Ai\Exceptions\AiAssistantArchivedException;
+
+it('completes the last response in a thread', function () {
+    asSuperAdmin();
+
+    $responseContent = AiMessage::factory()->make()->content;
+
+    /** @phpstan-ignore-next-line */
+    $this->mock(
```
Comment (Orrison) [inline app-modules/ai/tests/Feature/Http/Controllers/CompleteResponseControllerTest.php:58]: Isn't there a Pest `mock` function you could import and use here so you don't have to use `$this` and ignore the phpstan issue?

**Review by Orrison — CHANGES_REQUESTED** [review]
Looks good, just a couple of small nits.

File: app-modules/ai/tests/Feature/Http/Controllers/CompleteResponseControllerTest.php (hunk 2 above)
Comment (danharrin) [reply app-modules/ai/tests/Feature/Http/Controllers/CompleteResponseControllerTest.php:58]: I have tried, and there doesn't seem to be. `$this->mock()` on the TestCase actually binds the mocked instance to the Laravel container

```php
$this->instance($abstract, Mockery::mock(...array_filter(func_get_args())))
```

Whereas PEST's integration seems to be framework agnostic and not bind

**Review by Orrison — APPROVED** [review]

**Pull request merged**